# Tarea 2 - Portafolio de Inversión

**Integrantes:**  
María Fernanda Gómez Flores  
Carlos Hasan Hernández Rohan

**Periodo de análisis:** 01/01/2020 - 31/08/2026


In [1]:
#importar librerias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import yfinance as yf

## Selección de activos

Se seleccionaron los activos BDRY, FXY, WEAT y XLP.
- BDRY representa fletes marítimos.
- FXY representa el yen japonés.
- WEAT representa el mercado del trigo.
- XLP representa empresas de consumo básico.

Se escogieron porque pertenecen a mercados distintos, lo que ayuda a que el portafolio tenga una mejor diversificación.

In [2]:
#Definir activos
tickers=['BDRY','FXY','WEAT','XLP']

In [3]:
#descargar datos
prices=yf.download(tickers,start='2020-01-01',end='2026-09-01')["Close"]
prices

/var/folders/x9/60hcc2ln62d3r3xyv90d3wl40000gn/T/ipykernel_866/60511789.py:2: FutureWarning: YF.download() has changed argument auto_adjust default to True
  prices=yf.download(tickers,start='2020-01-01',end='2026-09-01')["Close"]
[*********************100%***********************]  4 of 4 completed


Ticker,BDRY,FXY,WEAT,XLP
Date,,,,
2020-01-02,13.961,87.510002,29.299999,52.624794
2020-01-03,13.050,87.919998,29.049999,52.540565
2020-01-06,11.920,87.650002,28.950001,52.650051
2020-01-07,11.895,87.510002,28.900000,52.245766
2020-01-08,13.655,87.099998,29.000000,52.439495
...,...,...,...,...
2026-08-25,15.065,57.630001,25.629999,86.519997
2026-08-26,15.060,57.549999,26.990000,86.269997
2026-08-27,14.820,57.500000,27.219999,85.080002


In [4]:
#rend diarios
rets=(prices/prices.shift()-1).dropna()
rets

Ticker,BDRY,FXY,WEAT,XLP
Date,,,,
2020-01-03,-0.065253,0.004685,-0.008532,-0.001601
2020-01-06,-0.086590,-0.003071,-0.003442,0.002084
2020-01-07,-0.002097,-0.001597,-0.001727,-0.007679
2020-01-08,0.147961,-0.004685,0.003460,0.003708
2020-01-09,-0.025265,-0.003444,0.015517,0.007067
...,...,...,...,...
2026-08-25,0.025528,-0.000173,0.005887,-0.010635
2026-08-26,-0.000332,-0.001388,0.053063,-0.002890
2026-08-27,-0.015936,-0.000869,0.008522,-0.013794


## Rendimiento diario y anual del portafolio

In [5]:
#Rendimiento esperado de cada activo
mu=rets.mean()

In [6]:
#peso de los activos en el portafolio
w_port=np.array([0.25,0.25,0.25,0.25])

In [7]:
#Rendimiento del Portafolio
r_port=sum(w_port*mu)
r_port

0.0002842889279028767

In [8]:
#rend promedio
mu=rets.mean()*252
mu*100

Ticker
BDRY    22.570629
FXY     -5.876365
WEAT     3.398575
XLP      8.563485
dtype: float64

In [9]:
#Rendimiento del Portafolio
r_port=sum(w_port*mu)
r_port

#es la suma ponderada del peso que tiene el activo en el portafolio por su rendimiento

0.07164080983152493

## Varianza diaria y anual del portafolio

In [10]:
#Riesgo portafolio
variance=rets.var()
variance

Ticker
BDRY    0.001681
FXY     0.000038
WEAT    0.000335
XLP     0.000107
dtype: float64

In [11]:
cov=rets.cov()
cov

Ticker,BDRY,FXY,WEAT,XLP
Ticker,,,,
BDRY,0.001681,2.483656e-06,-0.000010,2.243707e-06
FXY,0.000002,3.800721e-05,0.000005,-7.221274e-08
WEAT,-0.000010,4.503063e-06,0.000335,2.043040e-06
XLP,0.000002,-7.221274e-08,0.000002,1.065599e-04


In [12]:
daily_variance=w_port.T @ cov @ w_port

In [13]:
daily_variance

np.float64(0.00013520629998888168)

In [14]:
annual_variance=daily_variance*252
annual_variance*100

np.float64(3.4071987597198183)

## Volatilidad diaria y anual del portafolio

In [15]:
daily_volatility=np.sqrt(w_port.T @ cov @ w_port)

In [16]:
daily_volatility*100

np.float64(1.1627824387600703)

In [17]:
volatility_annual=daily_volatility*np.sqrt(252)
volatility_annual*100

np.float64(18.458598970994032)

## Matriz de correlación

In [18]:
#Matriz de correlación
rets.corr()

Ticker,BDRY,FXY,WEAT,XLP
Ticker,,,,
BDRY,1.000000,0.009826,-0.013279,0.005301
FXY,0.009826,1.000000,0.039895,-0.001135
WEAT,-0.013279,0.039895,1.000000,0.010810
XLP,0.005301,-0.001135,0.010810,1.000000


### Conclusión sobre la matriz de correlación y el riesgo no sistemático

Lo primero que nos llamó la atención de la matriz es que practicamente todas las correlaciones quedaron muy cercanas a 0, entonces los cuatro activos casi no se movieron de la misma manera durante el periodo que analizamos. Por ejemplo BDRY y XLP tuvieron una correlación de solamente 0.0053 y FXY con XLP de -0.0011, incluso la correlación mas alta de toda la matriz fue FXY con WEAT y fue de apenas 0.0399, entonces realmente sigue siendo bastante baja.

Esto tiene sentido por los activos que escogimos porque aunque todos forman parte del mismo portafolio, cada uno depende de cosas muy diferentes. BDRY esta relacionado con los fletes marítimos, FXY con el yen japonés, WEAT con el trigo y XLP con empresas de consumo básico. Entonces si por ejemplo pasa algo que afecte especificamente al mercado del trigo no necesariamente tendría que afectar de la misma forma al yen o a las empresas de consumo básico.

En cuanto al riesgo no sistemático, cada activo tambien tiene riesgos propios. BDRY puede verse afectado por cambios en los fletes marítimos y en la demanda de transporte, FXY por movimientos propios del yen japonés, WEAT por factores como la producción de trigo, el clima o cambios en la oferta agrícola y XLP por problemas específicos de las empresas del sector de consumo básico. En nuestros resultados BDRY fue el activo con mayor varianza y FXY el de menor, mientras que WEAT y XLP quedaron en medio.

Esto no quiere decir que el riesgo especifico de cada activo desaparezca al meterlo al portafolio, mas bien al combinarlos con activos que casi no estan correlacionados entre si se puede disminuir el efecto que tendría un problema de uno solo sobre todo el portafolio. Por eso creemos que en este caso la diversificación si ayuda bastante a reducir el riesgo no sistemático, porque los riesgos vienen de fuentes diferentes y los movimientos de los cuatro activos casi no se repiten.

### Conclusión sobre las métricas de rendimiento, riesgo y diversificación del portafolio

El portafolio tuvo un rendimiento diario promedio de aproximadamente 0.0284% y un rendimiento anual de 7.16%. La volatilidad diaria fue de 1.16% y la anual de 18.46%, entonces aunque el rendimiento promedio fue positivo también podemos ver que si hubo bastante variación en los rendimientos durante el periodo que analizamos.

La varianza diaria fue de 0.000135 y la anual de aproximadamente 0.0341. Creemos que aqui no solamente importa ver que el portafolio dio un rendimiento de 7.16%, porque si vemos solo eso realmente no estamos tomando en cuenta el riesgo. La volatilidad anual fue de 18.46%, entonces si hubo movimientos importantes aunque al final el rendimiento promedio haya sido positivo.

Para nosotros lo mas importante de este portafolio fue la diversificación, porque las correlaciones entre los cuatro activos quedaron practicamente en 0. Esto quiere decir que no se estuvieron moviendo todos de la misma manera y entonces el portafolio no depende tanto de lo que pase en un solo mercado. Por ejemplo si algo afecta al trigo no necesariamente va a afectar igual al yen, a los fletes marítimos o a las empresas de consumo básico.

Entonces creemos que el portafolio si cumple bien con lo que buscábamos desde el inicio, porque obtuvimos un rendimiento positivo pero además logramos combinar activos que casi no estan relacionados entre ellos. No es que por estar diversificado ya no exista riesgo, mas bien la idea es que el riesgo especifico de un activo no afecte de la misma forma a todo el portafolio y por eso creemos que la diversificación que obtuvimos fue bastante buena.